src/mavsdk/core/mavlink_ftp_client.h

```diff
@@ -151,12 +152,13 @@ class MavlinkFtpClient {
         std::string local_folder{};
         DownloadCallback callback{};
         std::ofstream ofstream{};
-        enum class Transferred {
-            No,
-            Yes,
+        uint32_t file_size{0};
+        struct MissingData {
+            size_t offset;
+            size_t size;
         };
-        std::vector<Transferred> transferred{};
-        int last_progress_percentage{-1};
+        std::list<MissingData> missing_data{};
```

Comment:
I would generally just use a `std::vector` unles you have strong reasons. I guess your thinking is that we are pushing at the back and then popping from the front? Actually, I think that should be a `std::queue`, right?

Reply:
Yes you're right